apps/server/src/shared/domain/entity/user.entity.ts

```diff
@@ -157,44 +158,69 @@ export class User extends BaseEntityWithTimestamps implements EntityWithSchool {
 			permissions = [...permissions, ...rolePermissions];
 		});
 
-		const setOfPermission = this.resolveSchoolPermissions(permissions, roles);
+		const setOfPermissions = this.resolveSchoolPermissions(permissions, roles);
 
-		const uniquePermissions = [...setOfPermission];
+		const uniquePermissions = [...setOfPermissions];
 
 		return uniquePermissions;
 	}
 
-	private resolveSchoolPermissions(permissions: string[], roles: Role[]) {
-		const setOfPermission = new Set(permissions);
+	private resolveSchoolPermissions(permissions: string[], roles: Role[]): Set<string> {
+		if (!wrap(this.school).isInitialized()) {
+			throw new ReferenceNotPopulatedLoggableException('user', 'school');
+		}
+
 		const schoolPermissions = this.school.permissions;
+		let setOfPermissions = new Set(permissions);
 
 		if (roles.some((role) => role.name === RoleName.ADMINISTRATOR)) {
```

Comment:
Mit RoleName wollen wir eigentlich auch nicht arbeiten. Müsste man dann vermutlich zu einem späteren Zeitpunkt auflösen. Wenn für alle Rollen return setOfPermission gilt, außer für teacher und student dann könnten die if Bedinung für den Administrator raus. Für Supherhero, Expert etc.. verhält es sich eh gleich.

Reply:
Das mit den Rollen muss hier so sein, weil Doppelrollen möglich sind. Einer der Bugs im Nachgang der Me-Migration war, dass Admins Permissions entzogen wurden, wenn sie auch Teacher sind. Daher muss die Admin-Rolle erst ausgeschlossen werden. 

Was mit Superhero und Expert ist, ist ne berechtigte Frage. Beim Superhero dürfte die fachliche Anforderung ziemlich klar sein, dass keine Permissions entzogen werden. Der Code ging hier einfach davon aus, dass kein Superhero ne Doppelrolle hat. Ich hab den Superhero mal mit ausgeschlossen. Das sollte jedenfalls nicht schaden.

Wie es sich verhalten soll, wenn Experts Doppelrollen haben, das find ich schwerer einzuschätzen. Ich denke, das müssten die POs klären.

Reply:
Ah ok, zu Experts kann zumindest direkt bestätigen das die nie eine Doppelrolle haben. Bei superhero sollte das eigentlich auch der Fall sein. Nur wäre es über shd möglich das anders zu machen.

Reply:
Der Vollständigkeit halber, wie eben schon mündlich besprochen: Die Permission steht für die Admin-Rolle in den Daten.